src/resolvers/coordinates.ts

```diff
@@ -0,0 +1,54 @@
+// import { log } from '.';
+// import { UserInputError } from 'apollo-server';
+import { Context } from 'apollo-server-core';
+import { getCoordinates } from '../calculateCoordinates';
+import conf from '../conf';
+import { GraphQLResolveInfo } from 'graphql';
+
+// Resolvers define the technique for fetching the types defined in the
+// schema.
+// I read a doc, where each resolver making a different async call would be better compared
+// to making one single async call
+// because failing to get longitude should not impact latitude
```

Comment:
Wouldn't this double the consumption of the Google Geocode API for each request?

Reply:
Yes right, but what 
1. if we just decided to use only longitude or latitude, 
2. what if making one async call fails for fetching longitude, which impacts latitude
By the way, this was the [document](https://medium.com/paypal-tech/graphql-resolvers-best-practices-cd36fdbcef55) I referred, can read once more (as it's been a while) to give some more explanation

Reply:
In this particular case, both longitude and latitude are fetched from the same data source. I would certainly agree with your decision if each field is obtained from a different source since in that case we would be optimizing for performance and independent retrieval of fields

Reply:
I can make this change `consuming API just once`

Reply:
I have finished implementing this by leveraging express context and passing this context to ApolloServer instance, and it is working. Now i am just trying to use this context for dataSources, and all dataSources i saw [here](https://www.apollographql.com/docs/apollo-server/data/data-sources/) doesn't align with my `getCoordinates` method ( because i am not making API call/talking to DB), so just trying to create a class with this method and see if i can make it work with dataSources.

```context: expressContextBuilder,```
```export const expressContextBuilder = async ({
  event,
  context,
  express,
}: any) => {
 console.log('express req:', express?.req.body?.variables.name);
  const address = await getCoordinates(
    express?.req.body?.variables.name,
    conf.apiKey
  );
  console.log('address in resolver:', address);
  return { address: address[0] };
};```

Reply:
Ok, I have used DataSources to consume Google Geocoder API once and pushed those changes just now, able to hit API `https://dqe0uhf4s5.execute-api.us-east-1.amazonaws.com/prod/graphql` fine after making these changes.
I need to do some cleanup and fix tests according to my recent changes, which i can do later today or tomorrow. Thanks for the helpful comments/review.

Reply:
Alright fixed tests, also added better error handling for jwt errors and tests for this.